session/session.go

```diff
@@ -0,0 +1,75 @@
+package session
+
+import (
+	"context"
+	"encoding/gob"
+	"fmt"
+	"net/http"
+
+	"github.com/gofrs/uuid/v5"
+	"github.com/gorilla/sessions"
+)
+
+var ctxKey contextKey = "session"
+
+type contextKey string
+
+func init() {
+	// TODO: Look for a better place
+	gob.Register(uuid.UUID{})
+}
+
+type Session interface {
+	Register(w http.ResponseWriter, r *http.Request)
+}
+
+func New(secret, name string, options ...Option) Session {
+	store := sessions.NewCookieStore([]byte(secret))
+
+	// Run the options on the store
+	for _, option := range options {
+		option(store)
+	}
+
+	return &session{
+		name:  name,
+		store: store,
+	}
+}
+
+type session struct {
+	name  string
+	store *sessions.CookieStore
+}
+
+func (s *session) Register(w http.ResponseWriter, r *http.Request) {
+	session := s.session(r)
+
+	*r = *r.WithContext(context.WithValue(r.Context(), ctxKey, session))
```

Comment:
@tatang26 shouldn't this should be r = r.WithContext...

Reply:
You're right, that is an unneeded change. I'll change it